### PR TITLE
Fix resolve extensions in modules

### PR DESCRIPTION
**Pull request by dalcib**
Extract extension from pkg.main before loadAsFileSync otherwise the opts.extensions are not resolved.

### CODE REVIEW

File: lib/sync.js [hunk 1]
```diff
@@ -68,6 +68,8 @@ module.exports = function (x, options) {
                 if (pkg.main) {
                     if (pkg.main === '.' || pkg.main === './') {
                         pkg.main = 'index';
+                    } else {
+                        pkg.main = pkg.main.match(/(.+?)\.[^.]*$|$/)[1]
```
Comment (ljharb) [inline lib/sync.js]: Can you provide a test that fails without your change? It's not clear to me what this is attempting to address.

**Comment by dalcib** [comment]
The test above try to attempt the situation where there is a package with two entry points, `index.js` and `index.coffee.js`, and the pkg.main points to `index.js`.
The desirable return is `index.coffee` with `[".coffee.js", ".js"]` but `index.js` is returned.
The concrete case is React-Navigation, where there is `index.js`and `index.web.js` files.
In react-native-web projects there will be other cases similar.

**Comment by dalcib** [comment]
I resolved the problem using `s`, but I think it is not the best solution.

```
var modulePath = resolve.sync(sourcePath, {
    basedir: path.resolve(currentFile, '..'),
    extensions: ['.web.js', '.js'],
    packageFilter: function(pkg) {
      if (pkg.main) {
        pkg.main = pkg.main.match(/(.+?)\.[^.]*$|$/)[1] || pkg.main
      }
      return pkg
    },
  })
```

File: test/resolver/bir/node_modules/bar/package.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "main": "index.js"
```
Comment (ljharb) [inline test/resolver/bir/node_modules/bar/package.json]: In this case, the "main" being "index.js" is why it's failing. If the "main" is "index", or ".", or "./", then the extension resolution will kick in.

I've made this change and pushed it up, so you can see that the tests pass.

**Comment by ljharb** [comment]
If the package main points to "index.js", that can *never* be pointed to any other extension. The only way "extensions" kicks in is if the "main" omits an extension.

**Comment by dalcib** [comment]
Ok, I understand.

**Pull request closed without merging**